mlflow/tracing/provider.py

```diff
@@ -215,15 +215,26 @@ def f():
 
     @functools.wraps(f)
     def wrapper(*args, **kwargs):
-        was_trace_enabled = _is_enabled()
-        if was_trace_enabled:
-            disable()
-            try:
+        _is_func_called = False
+        try:
+            was_trace_enabled = _is_enabled()
+            if was_trace_enabled:
+                disable()
+                try:
+                    _is_func_called = True
+                    return f(*args, **kwargs)
+                finally:
+                    enable()
+            else:
+                _is_func_called = True
                 return f(*args, **kwargs)
-            finally:
-                enable()
-        else:
-            return f(*args, **kwargs)
+        except Exception:
```

Comment:
Do we want to insert a logger here for debugging when something unexpected happens?

```suggestion
        except Exception:
            _logger.debug("...", exc_info=True)
```